homeassistant/components/elmax/__init__.py

```diff
@@ -0,0 +1,62 @@
+"""The elmax-cloud integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .common import ElmaxCoordinator
+from .const import (
+    CONF_ELMAX_PANEL_ID,
+    CONF_ELMAX_PANEL_PIN,
+    CONF_ELMAX_PASSWORD,
+    CONF_ELMAX_USERNAME,
+    DOMAIN,
+    ELMAX_PLATFORMS,
+    POLLING_SECONDS,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistantType, entry: ConfigEntry) -> bool:
+    """Set up elmax-cloud from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Create the API client object and attempt a login, so that we immediately know
+    # if there is something wrong with user credentials
+    coordinator = ElmaxCoordinator(
+        hass=hass,
+        logger=_LOGGER,
+        username=entry.data[CONF_ELMAX_USERNAME],
+        password=entry.data[CONF_ELMAX_PASSWORD],
+        panel_id=entry.data[CONF_ELMAX_PANEL_ID],
+        panel_pin=entry.data[CONF_ELMAX_PANEL_PIN],
+        name=f"Elmax Cloud {entry.entry_id}",
+        update_interval=timedelta(seconds=POLLING_SECONDS),
+    )
+
+    # Issue a first refresh, so that we trigger a re-auth flow if necessary
+    await coordinator.async_config_entry_first_refresh()
+
+    # Store a global reference to the coordinator for later use
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    # Perform platform initialization.
+    # Add the cover feature only if supported by the current panel.
+    plats = ELMAX_PLATFORMS.copy()
+    if coordinator.panel_status is not None and coordinator.panel_status.cover_feature:
+        plats.append("cover")
```

Comment:
There is no cover platform in your integration.

```suggestion
```

Reply:
Good spot: it did have it in the initial PR. Now it has not (due to single-platform best practice for first integrations). I've removed it, I'll add it back as soon as merging the cover platform.